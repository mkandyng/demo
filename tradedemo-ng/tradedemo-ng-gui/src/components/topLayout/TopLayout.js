import React from "react";
import {connect} from "react-redux";
import { bindActionCreators } from "redux";
import { placeOrder, updateOrder } from "../../store/orderbook/orderbookActions"
import { fetchInstruments } from "../../store/instruments/instrumentsActions";
import { addInstrumentToMarketfeed,
         selectInstrumentToMarketfeed,
         updateInstrumentToMarketfeed,
         deleteInstrumentToMarketfeed } from "../../store/marketfeed/marketfeedActions";
import Ticket from "../../components/ticket/Ticket";
import InstrumentSearch from "../../components/instrumentSearch/InstrumentSearch";
import Marketfeed from "../../components/marketfeed/Marketfeed";
import "./topLayout.css"


/**
 * Container for the bottomLayout components
 */

function TopLayout(props) {
  return (
      <div id="topLayout">
          <Ticket {...props} />
          <div id="marketfeed">
            <InstrumentSearch {...props} />
            <Marketfeed {...props}/>
          </div>
      </div>
  )
}

const mapStateToProps = state => ({ instruments: state.instruments,
                                    instrument: state.marketfeed.selected,
                                    marketfeed: state.marketfeed,
                                  });

// Map Redux actions to component props
const mapDispatchToProps = dispatch =>
    bindActionCreators({
      placeOrder,
      updateOrder,
      addInstrumentToMarketfeed,
      selectInstrumentToMarketfeed,
      updateInstrumentToMarketfeed,
      deleteInstrumentToMarketfeed,
      fetchInstruments,
    }, dispatch);

// The HOC
export default connect(mapStateToProps, mapDispatchToProps)(TopLayout);
