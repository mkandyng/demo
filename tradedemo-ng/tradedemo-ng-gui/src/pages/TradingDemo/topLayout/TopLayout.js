import React from "react";
import Ticket from "../../../modules/ticket/components";
import InstrumentSearch from "../../../modules/instruments/components";
import Marketfeed from "../../../modules/marketfeed/components";
import "./topLayout.css"


/**
 * Container for the bottomLayout components
 */

export default function TopLayout(props) {
  return (
      <div id="topLayout">
          <Ticket />
          <div id="marketfeed">
            <InstrumentSearch />
            <Marketfeed/>
          </div>
      </div>
  )
}
