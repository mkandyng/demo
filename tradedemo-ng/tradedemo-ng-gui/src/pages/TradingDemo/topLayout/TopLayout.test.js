import React from "react";
import toJson from "enzyme-to-json";
import {shallow} from "enzyme";
import {TopLayout} from "./TopLayout";

/**
 * This is unit test of TopLayout component
 * It leverage snapshot to test the presentation
 */

describe("TopLayout", () => {
  const props = {
    searchInstruments: [
      {
        symbol: "symbol"
      }
    ],
    marketfeedInstruments: [
      {
        symbol: "symbol"
      }
    ],
    ticket: {
      symbol: "symbol"
    },
    instrument: {
      symbol: "symbol"
    },
    fetchInstruments: jest.fn(),
    placeOrder: jest.fn(),
    updateOrder: jest.fn(),
    addInstrumentToMarketfeed: jest.fn(),
    selectMarketfeedInstrument: jest.fn(),
    updateMarketfeedInstrument: jest.fn(),
    deleteMarketfeedInstrument: jest.fn()
  }

  it("should render component comparing with previous snapshot shallow markup", () => {
    // Given

    // When
    const component = shallow(<TopLayout {...props}/>);

    // Then
    expect(toJson(component)).toMatchSnapshot();
  });

});
