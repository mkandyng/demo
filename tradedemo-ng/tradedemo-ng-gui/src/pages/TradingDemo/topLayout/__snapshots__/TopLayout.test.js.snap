// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopLayout should render component comparing with previous snapshot shallow markup 1`] = `
<div
  id="topLayout"
>
  <Ticket
    addInstrumentToMarketfeed={[MockFunction]}
    deleteMarketfeedInstrument={[MockFunction]}
    fetchInstruments={[MockFunction]}
    instrument={
      Object {
        "symbol": "symbol",
      }
    }
    marketfeedInstruments={
      Array [
        Object {
          "symbol": "symbol",
        },
      ]
    }
    placeOrder={[MockFunction]}
    searchInstruments={
      Array [
        Object {
          "symbol": "symbol",
        },
      ]
    }
    selectMarketfeedInstrument={[MockFunction]}
    ticket={
      Object {
        "symbol": "symbol",
      }
    }
    updateMarketfeedInstrument={[MockFunction]}
    updateOrder={[MockFunction]}
  />
  <div
    id="instrumentsMarketfeed"
  >
    <InstrumentsSearch
      addInstrumentToMarketfeed={[MockFunction]}
      deleteMarketfeedInstrument={[MockFunction]}
      fetchInstruments={[MockFunction]}
      instrument={
        Object {
          "symbol": "symbol",
        }
      }
      marketfeedInstruments={
        Array [
          Object {
            "symbol": "symbol",
          },
        ]
      }
      placeOrder={[MockFunction]}
      searchInstruments={
        Array [
          Object {
            "symbol": "symbol",
          },
        ]
      }
      selectMarketfeedInstrument={[MockFunction]}
      ticket={
        Object {
          "symbol": "symbol",
        }
      }
      updateMarketfeedInstrument={[MockFunction]}
      updateOrder={[MockFunction]}
    />
    <InstrumentsMarketfeed
      addInstrumentToMarketfeed={[MockFunction]}
      deleteMarketfeedInstrument={[MockFunction]}
      fetchInstruments={[MockFunction]}
      instrument={
        Object {
          "symbol": "symbol",
        }
      }
      marketfeedInstruments={
        Array [
          Object {
            "symbol": "symbol",
          },
        ]
      }
      placeOrder={[MockFunction]}
      searchInstruments={
        Array [
          Object {
            "symbol": "symbol",
          },
        ]
      }
      selectMarketfeedInstrument={[MockFunction]}
      ticket={
        Object {
          "symbol": "symbol",
        }
      }
      updateMarketfeedInstrument={[MockFunction]}
      updateOrder={[MockFunction]}
    />
  </div>
</div>
`;
